leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java

```diff
@@ -29,23 +36,116 @@
  */
 public class WriteRequest extends AbstractDownlinkRequest<WriteResponse> {
 
+    public enum Mode {
+        REPLACE, UPDATE
+    }
+
     private final LwM2mNode node;
     private final ContentFormat contentFormat;
+    private final Mode mode;
+
+    // ***************** write object ****************** //
+
+    public WriteRequest(final Mode mode, final ContentFormat contentFormat, final int objectId,
```

Comment:
I would remove the ContentFormat from the constructor and let the user to use a setter for forcing JSON or anything else after the construction.

This would remove ~5 signatures for the constructor but make the class mutable.

@sophokles73 WDYT?